dev/doc/release-process.md

```diff
@@ -25,7 +25,9 @@ in time.
   the next branch point with the `--release` flag (see next section).
 - [ ] Put the corresponding alpha tag using `git tag -s`.
   The `VX.X+alpha` tag marks the first commit to be in `master` and not in the
-  branch of the previous version.
+  branch of the previous version. Note that you should tag a real commit,
+  not a merge one. After tagging double check that `git describe` picks up
```

Comment:
Looks good, I'd maybe write "note that usually, the first commit on the new branch will not be the merge commit but the first commit of the first merged PR after the branching"